### PR TITLE
Add tools to Cargo workspace

### PR DESCRIPTION
**Pull request by BD103**
This adds all of the tools written in Rust to a [Cargo workspace], which centralizes `Cargo.lock`, `target/`, lints, and other configuration.

[Cargo workspace]: https://doc.rust-lang.org/cargo/reference/workspaces.html

Originally discussed [on Discord](https://discord.com/channels/691052431525675048/695741366520512563/1208202278046015508).

### CODE REVIEW

File: generate-release/.gitignore [hunk 1]
```diff
@@ -1,7 +1,4 @@
-target
-assets
-data/
```
Comment (BD103) [inline generate-release/.gitignore:3]: These folders don't seem to be used in `generate-release`. Perhaps this was copy-and-pasted from somewhere else?

**Comment by BD103** [comment]
Unfortunately CI is failing due to #977, so this is temporarily blocked. All that's left here is fixing the Clippy lints, updating CI, and updating any documentation.

**Comment by BD103** [comment]
I think this is finished. There are a lot file changes, but they're all small and digestible. The mains changes are:

- Creating the workspace `Cargo.toml`
- Updating `.gitignore`s
- Consolidating `Cargo.lock`
- Fixing Clippy lints
- Updating CI

**Review by TrialDragon — APPROVED** [review]

**Review by alice-i-cecile — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
Nice stuff. Thanks for the great commit history; it made the review a lot easier in this case.

**Pull request merged**